python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+        sncosmoModel = sncosmo.Model(source=snState['ModelSource'],
+                                     effects=[dust, dust],
+                                     effect_names=['host', 'mw'],
+                                     effect_frames=['rest', 'obs'])
+
+        sncosmoParams = self.sncosmoParamDict(snState, sncosmoModel)
+        sncosmoParams['mwebv'] = snState['MWE(B-V)']
+        sncosmoModel.set(**sncosmoParams)
+        return sncosmoModel
+
+    def summary(self):
+        '''
+        summarizes the current state of the SNObject class in a returned
+        string.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Summary State in string format
+
+        Examples
+        --------
+        >>> t = SNObject()
+        >>> print t.summary()
+        '''
+        state = '  SNObject Summary      \n'
+
+        state += 'Model = ' + '\n'
+        state += 'z = ' + str(self.get('z')) + '\n'
+        state += 'c = ' + str(self.get('c')) + '\n'
+        state += 'x1 = ' + str(self.get('x1')) + '\n'
+        state += 'x0 = ' + str(self.get('x0')) + '\n'
+        state += 't0 = ' + str(self.get('t0')) + '\n'
+        state += 'ra = ' + str(self._ra) + ' in radians \n'
+        state += 'dec = ' + str(self._dec) + ' in radians \n'
+        state += 'MW E(B-V) = ' + str(self.ebvofMW) + '\n'
+
+        return state
+
+    def setCoords(self, ra, dec):
+        """
+        set the ra and dec coordinate of SNObject to values in radians
+        corresponding to the given values in degrees
+
+        Parameters
+        ----------
+        ra: float, mandatory
+            the ra in degrees
+        dec: float, mandatory
+            dec in degrees
+
+        Returns
+        -------
+        None
+
+        Examples
+        --------
+        >>> t = SNObject()
+        >>> t.setCoords(ra=30., dec=90.)
+        >>> t._ra
+        >>> 0.5235987755982988
+        >>> t._dec
+        >>> 1.0471975511965976
+        """
+
+        if ra is None or dec is None:
+            raise ValueError('Why try to set coordinates without full'
+                             'coordiantes?\n')
+        self._ra = np.radians(ra)
+        self._dec = np.radians(dec)
+        self.skycoord = np.array([[self._ra], [self._dec]])
+
+        self._hascoords = True
+
+        return
+
+    def set_MWebv(self, value):
+        """
+        if mwebv value is known, this can be used to set the attribute
+        ebvofMW of the SNObject class to the value (float).
+
+        Parameters
+        ----------
+        value: float, mandatory
+               value of mw extinction parameter E(B-V) in mags to be used in
+               applying extinction to the SNObject spectrum
+
+        Returns
+        -------
+        None
+
+        Examples
+        --------
+        >>> t = SNObject()
+        >>> t.set_MWebv(0.)
+        >>> 0.
+
+        .. note:: For a large set of SN, one may use fast `np.ndarray` valued
```

Comment:
Agreed. This seems to be wrong due to the reason you mention. I cannot remember any rationale for having that line. I am getting rid of it in 4b6b9814ea98e86f89517bd575794134388fcb24